extension/background.js

```diff
@@ -1,6 +1,44 @@
+let connections = {};
+let apolloConnected = false;
+let backgroundPageConnection = undefined;
+let requestLog = {};
+
+chrome.runtime.onConnect.addListener((port) => {
+  tabId = port.name
+  connections[tabId] = port;
+
+  port.onDisconnect.addListener(port => {
+
+    port.onMessage.removeListener(extensionListener);
+
+    const tabs = Object.keys(connections);
+    for (let i = 0; i < tabs.length; i++) {
+      if (connections[tabs[i]] == port) {
+        delete connections[tabs[i]];
```

Comment:
I think this entire for loop thingy can just be replaced with

```js
delete connections[tabId];
```

Or am I missing something?